### PR TITLE
Feat: Added copy colony address button to dashboard

### LINKED ISSUE #1344: Adding a copy Colony contract address to the dashboard
## Description

A copy icon and interaction should be added to the new dashboard to allow users to copy the Colony's contract address to their clipboard. The copy feature should also be added to the kebab menu dropdown. 

- Add in the copy icon to the main dashboard header as shown in Figma.
- Add a hover state of text-blue-900
- Add a copied tooltip interaction (utilise the same interaction as the share icon)
- Add in the dropdown menu as shown in Figma.
- Use the same copied tooltip state in the dropdown. 

## Figma links

Dashboard link with new icon - https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-V5?type=design&node-id=15919-214141&mode=design&t=6jWe0sq1QWEq9jhU-4

Dropdown with new menu item - https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-V5?type=design&node-id=15919-214606&mode=design&t=6jWe0sq1QWEq9jhU-4

### PR DESCRIPTION
**Pull request by iamsamgibbs**
## Description

Added copy colony address button to dashboard header and dropdown menu using the same interactions as the share url button.

Renamed the share url copyToClipboard functions to make them more specific.

**New stuff** ✨

* Added new copy colony address button to dashboard header which copies the colony address to the users clipboard when clicked
* Added new copy colony address button to dashboard dropdown menu which copies the colony address to the users clipboard when clicked

**Changes** 🏗

* Renamed the share url copyToClipboard functions to make them more specific.

Screenshot 2024-01-12 at 14 03 43

Screenshot 2024-01-12 at 14 03 55

Screenshot 2024-01-12 at 14 04 00

Screenshot 2024-01-12 at 14 04 04

Resolves #1344

### CODE REVIEW

File: src/components/common/ColonyHome/hooks/useDashboardHeader.ts [hunk 1]
```diff
@@ -44,17 +45,29 @@ export const useDashboardHeader = (): ColonyDashboardHeaderProps => {
   const { colony, colonySubscription } = useColonyContext();
   const { pathname } = useLocation();
   const colonyUrl = `${window.location.host}${pathname}`;
-  const { handleClipboardCopy, isCopied } = useCopyToClipboard(5000);
   const {
-    handleClipboardCopy: itemHandleClipboardCopy,
-    isCopied: itemIsCopied,
+    handleClipboardCopy: shareUrlhandleClipboardCopy,
```
Comment (bassgeta) [inline src/components/common/ColonyHome/hooks/useDashboardHeader.ts:49]: could we rename all of these to something like `handleShareUrlCopy`

**Review by bassgeta — CHANGES_REQUESTED** [review]
Nice job with the more granular copy to clipboard actions, just some syntax related comments dropped :D

File: src/components/common/ColonyHome/hooks/useDashboardHeader.ts (hunk 1 above)
Comment (iamsamgibbs) [reply src/components/common/ColonyHome/hooks/useDashboardHeader.ts]: Made this change! :)

**Review by bassgeta — DISMISSED** [review]

**Review by bassgeta — APPROVED** [review]

**Review by jakubcolony — APPROVED** [review]
LGTM, nice work! 👍

**Pull request merged**